bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+
+	// Disable Do not deploy packaged components and delete any deployed components (valid items: rke2-coredns, rke2-ingress-nginx, rke2-metrics-server)
+	// +optional
+	Disable []DisabledItem `json:"disable,omitempty"`
+
+	// DisableScheduler Disable Kubernetes default scheduler
+	// +optional
+	DisableScheduler string `json:"disable-scheduler,omitempty"`
+
+	// DisableCloudController Disable rke2 default cloud controller manager
+	// +optional
+	DisableCloudController string `json:"disableCloudController,omitempty"`
+
+	// DisableKubeProxy Disable running kube-proxy
+	// +optional
+	DisableKubeProxy string `json:"disableKubeProxy,omitempty"`
+
+	// NodeNamePrefix Prefix to be used by CAPI to generate the Node name
+	// +optional
+	NodeNamePrefix string `json:"nodeNamePrefix,omitempty"`
+
+	// NodeLabel Registering and starting kubelet with set of labels
+	// +optional
+	NodeLabels []string `json:"nodeLabel,omitempty"`
+
+	// NodeTaint Registering kubelet with set of taints
+	// +optional
+	NodeTaints []string `json:"nodeTaint,omitempty"`
+
+	// TODO: Remove ImageCredentialProviderBinDir when implementation is understood.
+	// ImageCredentialProviderBinDir The path to the directory where credential provider plugin binaries are located (default: "/var/lib/rancher/credentialprovider/bin")
+	// +optional
+	//ImageCredentialProviderBinDir string `json:"imageCredentialProviderBinDir,omitempty"`
+
+	// ImageCredentialProviderConfigMap is a reference to the ConfigMap that contains credential provider plugin config
+	// The configMap should contain a YAML file content + a Path to the Binaries for Credential Provider.
+	// +optional
+	ImageCredentialProviderConfigMap ConfigMapRef `json:"imageCredentialProviderConfigMap,omitempty"`
+
+	// TODO: Remove ContainerRuntimeEndpoint since this feature will probably not be offered by CAPI Bootstrap provider?
+
+	// ContainerRuntimeEndpoint Disable embedded containerd and use alternative CRI implementation
+	// +optional
+	ContainerRuntimeEndpoint string `json:"containerRuntimeEndpoint,omitempty"`
+
+	// Snapshotter Override default containerd snapshotter (default: "overlayfs")
+	// +optional
+	Snapshotter string `json:"snapshotter,omitempty"`
+
+	// TODO: Find a way to handle IP addresses that should be advertised but that RKE2 cannot find on the host (Example: Elastic IPs on Cloud Providers)
+
+	// NodeIp IPv4/IPv6 addresses to advertise for node
+	// +optional
+	//NodeIp string `json:"nodeIp,omitempty"`
+
+	// NodeExternalIp IPv4/IPv6 external IP addresses to advertise for node
+	// +optional
+	// NodeExternalIp string `json:"nodeExternalIp,omitempty"`
+
+	// ResolvConf Kubelet resolv.conf file
+	// +optional
+	ResolvConf string `json:"resolvConf,omitempty"`
+
+	// ProtectKernelDefaults Kernel tuning behavior. If set, error if kernel tunables are different than kubelet defaults.
+	// +optional
+	ProtectKernelDefaults string `json:"protectKernelDefaults,omitempty"`
```

Comment:
is this a bool flag?

Reply:
Good catch! Yes, it is supposed to be `bool`! Do you suggest an approach such as `ProtectKernelDefaultsPolicy` as an enum ?

Reply:
yes, thanks!

Reply:
Done!